cfme/tests/infrastructure/test_host.py

```diff
@@ -452,30 +450,28 @@ def test_infrastructure_hosts_navigation_after_download(
 
 @test_requirements.infra_hosts
 @pytest.mark.parametrize("num_hosts", [2, 4])
+@pytest.mark.parametrize("hosts_collection", ["provider", "appliance"])
 @pytest.mark.uncollectif(
     lambda provider, num_hosts: provider.one_of(RHEVMProvider, SCVMMProvider) and num_hosts > 2,
     reason=UNCOLLECT_REASON)
 @pytest.mark.meta(blockers=[BZ(1746214, forced_streams=["5.10"])], automates=[1746214, 1784181])
-def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_hosts, provider,
-                                              num_hosts):
+def test_infrastructure_hosts_compare(appliance, setup_provider_min_hosts, provider, num_hosts,
+                        hosts_collection):
     """
     Polarion:
         assignee: prichard
         casecomponent: Infra
         caseimportance: high
         initialEstimate: 1/6h
-    Bugzilla:
-        1746214
-
     """
-    ent_slice = slice(0, num_hosts, None)
-    hosts_view = navigate_to(provider.collections.hosts, "All")
-    for h in hosts_view.entities.get_all(slice=ent_slice):
-        h.ensure_checked()
-    hosts_view.toolbar.configuration.item_select('Compare Selected items',
-                                                 handle_alert=True)
-    compare_hosts_view = provider.create_view(ProviderHostsCompareView)
-    assert compare_hosts_view.is_displayed
+    if hosts_collection == 'provider':
+        h_coll = provider.hosts.all()[:num_hosts]
+        compare_view = provider.hosts.compare_entities(provider, entities_list=h_coll)
+        assert compare_view.is_displayed
+    elif hosts_collection == 'appliance':
+        h_coll = appliance.collections.hosts.all()[-num_hosts:]
```

Comment:
Why change the list slicing?

Reply:
Just to add another scenario to cover. We are always picking the first item. Thought we might should test using the last item returned.

Reply:
We don't do this any more. Same slices are used for iterations of the test.